runtime/filetype.vim

```diff
@@ -2897,6 +2897,9 @@ au BufNewFile,BufRead {.,}tmux*.conf*		setf tmux
 " Universal Scene Description
 au BufNewFile,BufRead *.usda,*.usd		setf usd
 
+" Uci
+au BufNewFile,BufReadPost */etc/config/*		setf uci
+
```

Comment:
```suggestion
```

I think this would lead to UCI being used for a lot of files it shouldn't be.

Reply:
agree, that looks wrong. Is there a better pattern for uci files?

Reply:
I don't think there's any better way of matching on their paths.

Paths to UCI files:
https://openwrt.org/docs/guide-user/base-system/uci#configuration_files

This pattern matches all the UCI files, and rarely matches other files in my use, which may be good enough, but I'm sure it will match files that it is not intended to.

Maybe that's fine. I don't know what the baseline is here or what the consequences are for plugins if vim starts assigning the wrong filetype here.

Reply:
I'm not sure this is resolved yet. Can someone confirm that it's okay for filetypes.vim to risk false positives, even for obscure filetypes like UCI?

Reply:
I think the best way is let upstream use some path like `/etc/uci-config/`?

Reply:
The files are all in the /etc/config directory, and I matched */etc so that accessing an openwrt filesystem mounted on e.g. sshfs will still work.

The filenames don't have a matchable pattern, imo.

> I think the best way is let upstream use some path like /etc/uci-config/?

I don't think openwrt are gonna change their config directory name to accommodate this plugin :)

----

Could someone respond to this question I asked?

> Can someone confirm that it's okay for filetypes.vim to risk false positives, even for obscure filetypes like UCI?

Reply:
Right, sorry for misunderstanding you!

I didn't know filetype.vim was allowed to match on file contents :)

For what it's worth I'm okay with the filepath pattern I originally had, and I'm also okay if the file contents are also scanned, but I don't know how to do that :)

Anyway, I'm gonna peace out of this issue now. Tag me again if I'm needed!

Reply:
@cmcaine please check dist/ft.vim they mostly are all there for content detection.